apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java

```diff
@@ -140,6 +143,31 @@ public Uni<Onboarding> persistAndStartOrchestrationOnboarding(Onboarding onboard
         }
     }
 
+    private WorkflowType getWorkflowType(Onboarding onboarding) {
+        InstitutionType institutionType = onboarding.getInstitution().getInstitutionType();
+        if(InstitutionType.PT.equals(institutionType)){
+            return WorkflowType.FOR_APPROVE_PT;
+        }
+
+        if(InstitutionType.PA.equals(institutionType)
+                || checkIfGspProdInterop(institutionType, onboarding.getProductId())
+                || InstitutionType.SA.equals(institutionType)
+                || InstitutionType.AS.equals(institutionType)) {
+            return WorkflowType.CONTRACT_REGISTRATION;
+        }
+
+        if(InstitutionType.PG.equals(institutionType)) {
+            return WorkflowType.CONFIRMATION;
+        }
+
+        return WorkflowType.FOR_APPROVE;
+    }
+
+    private boolean checkIfGspProdInterop(InstitutionType institutionType, String productId) {
```

Comment:
naming: `isGspProdInterop` (I think we can make it clearer but I don't know the context to make a better suggestion). try to start every boolean method or value with `is` or `has` or `should`, it makes the code readable as an english phrase.